### PR TITLE
HAWQ-1593. Vectorized execution condition check in plan tree

### PR DESCRIPTION
**Pull request by zhangjackey**
1. Replace the vectorized data types and expressions immediately after check.
 2.Move the VExecInitNode to the end of the ExecInitNode.
 3.Process the Parent PlanState and the vectorized flag in the VExecInitNode.

### CODE REVIEW

File: contrib/vexecutor/vcheck.c [hunk 1]
```diff
@@ -54,7 +61,274 @@ typedef struct VecFuncHashEntry
...
+
+    ReplacePlanVectorzied(root, plan->lefttree);
+    ReplacePlanVectorzied(root, plan->righttree);
+    ReplacePlanNodeWalker(root, plan);
+
+    return plan;
+}
+
+Plan*
+CheckAndReplacePlanVectorized(PlannerInfo *root, Plan *plan)
```
Comment (wengyanqing) [inline contrib/vexecutor/vcheck.c]: How to check initPlans of main Plan node?

File: contrib/vexecutor/vcheck.c [hunk 2]
```diff
@@ -54,7 +61,274 @@ typedef struct VecFuncHashEntry
...
+
+        if(ctx->replace)
+        {
+        	    op->opresulttype = rettype;
+        	    op->opfuncid = voper->oprcode;
+        }
+
+        ctx->retType = rettype;
+        return false;
+    }
```
Comment (wengyanqing) [inline contrib/vexecutor/vcheck.c]: Maybe need to check more nodes, such as FuncExpr.

File: contrib/vexecutor/vcheck.c (hunk 1 above)
Comment (zhangjackey) [reply contrib/vexecutor/vcheck.c]: So far it is not supported, but it does not seem to affect the normal running.

File: contrib/vexecutor/vcheck.c (hunk 2 above)
Comment (zhangjackey) [reply contrib/vexecutor/vcheck.c]: Yes, we will support more expression nodes step by step.

**Comment by wengyanqing** [comment]
LGTM.

**Pull request closed without merging**